### PR TITLE
Fixing the prependExperiment method

### PR DESCRIPTION
**Pull request by bernardosrulzon**
**CHANGELOG** :memo:

* Fixing problems with the prepend method
* Update specs
* Update Docs
* Added Badges
* Added DroneCI support
* Removed Hound CI

### CODE REVIEW

File: dist/tagmanager-wrap.js [hunk 1]
```diff
@@ -100,7 +100,8 @@
     }, {
```
Comment (iondrimba) [inline dist/tagmanager-wrap.js:100]: Dist file, can be ignored 👍

File: index.html [hunk 2]
```diff
@@ -25,8 +25,8 @@ <h3>Init (ES6)</h3>
   <pre>
   <code class="language-javascript line-numbers">import TagManager from 'tagmanager-wrap';
 
-  window.tagManagerDatalayer = [];
-  const tagmanager = new TagManager(window.tagManagerDatalayer, {
+  window.tagManagerDataLayer = [];
```
Comment (iondrimba) [inline index.html]: Demo code

File: karma.conf.js [hunk 3]
```diff
@@ -45,13 +45,9 @@ const karmaConfig = function (conf) {
...
         type: 'text-summary',
       },
-      {
-        type: 'html',
-        subdir: 'report-html',
-      },
       {
         type: 'lcov',
-        subdir: 'report-lcov',
+        subdir: '/',
```
Comment (iondrimba) [inline karma.conf.js:50]: Changed coverage results output path for code climate integration

File: spec/tagManagerWrap.spec.js [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import TagManager from '../src/tagmanager-wrap';
 
 describe('Tagmanager', () => {
-  const tagManager = new TagManager([], {
+  window.tagManagerDataLayer = [];
```
Comment (iondrimba) [inline spec/tagManagerWrap.spec.js:4]: Using googles global `tagManagerDataLayer` as reference

File: src/tagmanager-wrap.js [hunk 5]
```diff
@@ -65,6 +66,10 @@ export default class TagManager {
     script.innerHTML = `(function (w, d, s, l, i) { w[l] = w[l] || []; w[l].push({ 'gtm.start': new Date().getTime(), event: 'gtm.js' }); var f = d.getElementsByTagName(s)[0], j = d.createElement(s), dl = l != 'dataLayer' ? '&l=' + l : ''; j.async = true; j.src = 'https://www.googletagmanager.com/gtm.js?id=' + i + dl; f.parentNode.insertBefore(j, f); })(window, document, 'script', 'tagManagerDataLayer', '${this.options.gtmId}');`;
 
     document.body.appendChild(script);
+
+    Object.assign(window.tagManagerDataLayer, this.dataLayer);
```
Comment (iondrimba) [inline src/tagmanager-wrap.js:71]: Merging local `dataLayer` ref with googles global `tagManagerDataLayer ` ref

File: README.md [hunk 6]
```diff
@@ -1,7 +1,11 @@
-# Tagmanager Wrap
+# TagManager Wrap
 
-Wrapping Google tagmanager and adding some helpers to him
-* This script appends automatically the googleTagmanager script.
+[![Build Status](https://semaphoreci.com/api/v1/iondrimba/tagmanager-wrap/branches/master/badge.svg)](https://semaphoreci.com/iondrimba/tagmanager-wrap)
```
Comment (iondrimba) [inline README.md]: Badges Semaphore, Code Climate

**Review by williamcosta — APPROVED** [review]

**Review by iondrimba — CHANGES_REQUESTED** [review]
There is some minor fixes to do 👍

File: src/tagmanager-wrap.js [hunk 7]
```diff
@@ -75,7 +81,8 @@ export default class TagManager {
     iframe.src = `https://www.googletagmanager.com/ns.html?id=${this.options.gtmId}`;
     iframe.height = 0;
     iframe.width = 0;
-    iframe.style = 'display:none;visibility:hidden';
+    iframe.style.display = 'none';
```
Comment (iondrimba) [inline src/tagmanager-wrap.js:84]: Changed this implementation to fix a Safari 10 problem

File: .drone.yml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+pipeline:
```
Comment (iondrimba) [inline .drone.yml:1]: Added build with drone CI, thanks @williamcosta 👍

File: karma.conf.js [hunk 9]
```diff
@@ -72,6 +68,7 @@ const karmaConfig = function (conf) {
         base: 'Chrome',
         flags: [
           '--headless',
+          '--no-sandbox',
```
Comment (iondrimba) [inline karma.conf.js:71]: Added this config to run Chrome inside drone CI

File: dist/tagmanager-wrap.js [hunk 10]
```diff
@@ -100,7 +100,9 @@
...
       key: 'prependExperiment',
       value: function prependExperiment(experiment) {
-        this.dataLayer[0].experiments.push(experiment);
+        this.options.startPush.experiments.push(experiment);
+
+        this.custom({ event: 'selfDescribingEvent', schema: 'iglu:br.com.getninjas/split_test/jsonschema/1-0-0', data: experiment });
```
Comment (felipeorlando) [inline dist/tagmanager-wrap.js]: The value for schema must be customizable.

Comment (iondrimba) [reply dist/tagmanager-wrap.js]: Nice catch! I will try to make `{ event: 'selfDescribingEvent', schema: 'iglu:br.com.getninjas/split_test/jsonschema/1-0-0', data: experiment }` it as an option config

**Review by felipeorlando — APPROVED** [review]

File: demo.js [hunk 11]
```diff
@@ -21,15 +20,20 @@ btnEventCategory.addEventListener('click', () => {
...
-    experimentId: 'popup-user-leaves-request',
-    experimentPageCategory: ['all'],
-    experimentPageType: ['categories'],
-    experimentTool: 'Abba',
-    experimentType: 'page',
-    experimentVersion: 'demo',
+    event: 'yourCustomEvent',
+    schema: 'your:br.com.custom/schema/jsonschema/1-0-0',
+    data: {
+      experimentDescription: 'PopUp/Modal quando o usuário deixa formulário',
```
Comment (eduardojmatos) [inline demo.js]: Can we use EN text here?

**Review by eduardojmatos — APPROVED** [review]

File: demo.js (hunk 11 above)
Comment (iondrimba) [reply demo.js]: Sure!

**Review by iondrimba — APPROVED** [review]

File: demo.js (hunk 11 above)
Comment (iondrimba) [reply demo.js]: DONE! @eduardojmatos  👍

**Review by williamcosta — APPROVED** [review]

**Pull request merged**